homeassistant/components/intellifire/sensor.py

```diff
@@ -0,0 +1,130 @@
+        [IntellifirePollData], int
+    ]  # Although sensours could have a variety of different return values - all the ones below are only reutrning ints
+
+
+@dataclass
+class IntellifireSensorEntityDescription(
+    SensorEntityDescription, IntellifireSensorEntityDescriptionMixin
+):
+    """Describes a sensor sensor entity."""
+
+
+ATTRIBUTION = "Data provided by unpublished Intellifire API"
+
+
+INTELLIFIRE_SENSORS: tuple[IntellifireSensorEntityDescription, ...] = (
+    IntellifireSensorEntityDescription(
+        key="flame_height",
+        icon="mdi:fire-circle",
+        name="Flame Height",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.flameheight,
+    ),
+    IntellifireSensorEntityDescription(
+        key="temperature",
+        icon="mdi:car-brake-temperature",
+        name="Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        value_fn=lambda data: data.temperature_c,
+    ),
+    IntellifireSensorEntityDescription(
+        key="target_temp",
+        icon="mdi:thermometer-lines",
+        name="Target Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        value_fn=lambda data: data.thermostat_setpoint_c,
+    ),
+    IntellifireSensorEntityDescription(
+        key="fan_speed",
+        icon="mdi:fan",
+        name="Fan Speed",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.fanspeed,
+    ),
+    IntellifireSensorEntityDescription(
+        key="timer_remaining",
+        icon="mdi:timer-sand",
+        name="Timer Time Remaining",
+        native_unit_of_measurement=TIME_MINUTES,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: round(int(data.timeremaining_s) / 60),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Define setup entry call."""
+
+    coordinator: IntellifireDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        IntellifireSensor(coordinator=coordinator, description=description)
+        for description in INTELLIFIRE_SENSORS
+    )
+
+
+class IntellifireSensor(CoordinatorEntity, SensorEntity):
+    """Define a generic class for Sensors."""
+
+    # Define types
+    coordinator: IntellifireDataUpdateCoordinator
+    entity_description: IntellifireSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: IntellifireDataUpdateCoordinator,
+        description: IntellifireSensorEntityDescription,
+    ) -> None:
+        """Init the sensor."""
+        super().__init__(coordinator=coordinator)
+        self.entity_description = description
+
+        self._attrs = {ATTR_ATTRIBUTION: ATTRIBUTION}
```

Comment:
This should be a class attribute (not instance attribute) and use `_attr_attribution`:
```python
class IntellifireSensor(CoordinatorEntity, SensorEntity):
    """Define a generic class for Sensors."""

    coordinator: IntellifireDataUpdateCoordinator
    entity_description: IntellifireSensorEntityDescription
    _attr_attribution = ATTRIBUTION
```

If it should be used on all platforms, maybe also move `ATTRIBUTION` to constants file.